planner/core/find_best_task.go

```diff
@@ -1130,12 +1148,18 @@ func (ds *DataSource) isSafePointGetPlan4PlanCache(path *util.AccessPath) bool {
 }
 
 func (ds *DataSource) convertToIndexMergeScan(prop *property.PhysicalProperty, candidate *candidatePath, _ *physicalOptimizeOp) (task task, err error) {
-	if prop.IsFlashProp() || prop.TaskTp == property.CopSingleReadTaskType || !prop.IsSortItemEmpty() {
+	if prop.IsFlashProp() || prop.TaskTp == property.CopSingleReadTaskType {
 		return invalidTask, nil
 	}
 	if prop.TaskTp == property.CopMultiReadTaskType && candidate.path.IndexMergeIsIntersection {
 		return invalidTask, nil
 	}
+	if !prop.IsSortItemEmpty() && !candidate.isMatchProp {
+		return invalidTask, nil
+	}
+	if candidate.isMatchProp && len(candidate.path.TableFilters) > 0 {
```

Comment:
Why can't there be `TableFilters`?

Reply:
If so, the limit cannot be added in partial paths, it can only be added to the final table path.

Reply:
Oh, there's a limitation: the codes in the executor indicate that when there's keep order. There must be a pushed limit.

Reply:
Fixed by introducing the new commit.